projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -54,8 +61,14 @@ def execute_dag(self):
         while self._topological_sorter.is_active():
             for node in self._topological_sorter.get_ready():
                 self._start_job(node)
-            self._start_delayed_jobs()
-
+            if self._delayed_starting_jobs.size() > 0:
```

Comment:
is this if needed?

Reply:
The if statement goes along with the while loop. Actually, I decided to use the if+while because, at least from what I saw, it performs better in terms of time - the number of operations are nearly 1/2 of the ones using the other approach. However, I agree that removing the if+while, the code would be cleaner. If optimality is not the high priority, should I remove it still?

Reply:
Performance is important here because we are sending a lot of requests. to the API: 

```
it performs better in terms of time - the number of operations are nearly 1/2 of the ones using the other approach
```

How do you measure it? What operations do you mean (API calls? -those are fairly expensive and reducing them is a bonus).

Reply:
Removed it.